twitter-clone/app/src/main/java/io/github/vnicius/twitterclone/data/datasource/usertweets/UserTweetsDataSource.kt

```diff
@@ -0,0 +1,67 @@
+package io.github.vnicius.twitterclone.data.datasource.usertweets
+
+import android.arch.lifecycle.MutableLiveData
+import android.arch.paging.PageKeyedDataSource
+import io.github.vnicius.twitterclone.data.repository.user.UserRepository
+import io.github.vnicius.twitterclone.utils.State
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import twitter4j.Status
+import java.lang.Exception
+
+class UserTweetsDataSource(
+    private val userId: Long,
+    private val itemsCount: Int,
+    private val userRepository: UserRepository
+) : PageKeyedDataSource<Int, Status>() {
+
+    private val userTweetsDataSourceJob = SupervisorJob()
+    private val userTweetsDataSourceScope = CoroutineScope(Dispatchers.IO + userTweetsDataSourceJob)
+    private var currentPage = 1
```

Comment:
You don't need to keep track of the current page. Instead use the `params.key`. In the loadInitial you can hardcode to fetch page number 1 and then check if you have a next to pass page 2 to the callback. Passing `null` to the adjacent page key means there is no more pages to fetch.